lib/chef/provider/package/powershell.rb

```diff
@@ -0,0 +1,122 @@
+          end
+        end
+
+        def remove_package(names, versions)
+          # Removes the package, if no version is passed, all installed version  will be removed
+          names.each_with_index do |name, index|
+            if versions && versions[index] != nil
+              powershell_out( "Uninstall-Package #{name} -Force -RequiredVersion #{versions[index]}", { :timeout => @new_resource.timeout })
+            else
+              stdout = "0"
+              until stdout.empty?
+                stdout = powershell_out( "Uninstall-Package #{name} -Force", { :timeout => @new_resource.timeout }).stdout
+                if !stdout.empty?
+                  Chef::Log.info("Removed package #{name} with version #{parse_version(stdout.downcase, name.downcase)}")
+                end
+              end
+            end
+          end
+        end
+
+        # return array of latest available packages online
+        def build_candidate_versions
+          versions = []
+          new_resource.package_name.each_with_index do |name, index|
+            if new_resource.version && new_resource.version[index] != nil
+              stdout = powershell_out("Find-Package #{name} -RequiredVersion #{new_resource.version[index]}", { :timeout => @new_resource.timeout }).stdout
+            else
+              stdout = powershell_out("Find-Package #{name}", { :timeout => @new_resource.timeout }).stdout
+            end
+            versions.push(parse_version(stdout.downcase, name.downcase))
+          end
+          versions
+        end
+
+        #return array of currently installed version
+        def build_current_versions
+          version_list = []
+          new_resource.package_name.each_with_index do |name, index|
+            if new_resource.version && new_resource.version[index] != nil
+              stdout = powershell_out("Get-Package -Name #{name} -RequiredVersion #{new_resource.version[index]}", { :timeout => @new_resource.timeout }).stdout
+            else
+              stdout = powershell_out("Get-Package -Name #{name}", { :timeout => @new_resource.timeout }).stdout
+            end
+            version_list.push(parse_version(stdout.downcase, name.downcase))
+          end
+          version_list
+        end
+
+        def parse_version(output, name)
+          if output.empty?
+            nil
+            #raise Chef::Exceptions::Package, "Invalid Package name specified or proper version not passed"
+          else
+            #sample value of output variable
+            #Name                           Version          Source           Summary
+            #----                           -------          ------           -------
+            #xCertificate                   2.1.0.0          PSGallery        This module includes DSC resources that simplify administration of certificates on a Windows Server
+            output_list = output.split(" ")
```

Comment:
I think this pattern of splitting on a space is going to be a problem. `Get-Package | Format-List` on my Windows 10 machine has this format:

```
Microsoft Visual C++ 2012 R... 11.0.51106.1                                      Programs
```

It looks like those might be tabs between the columns which you could split on, but in either case we're going to have matching problems on packages with spaces in their names and long package names.

```
PS C:\Users\btm_000\Downloads> Get-package -Name "Picture Control Utility x64"

Name                           Version          Source                           ProviderName
----                           -------          ------                           ------------
Picture Control Utility x64    1.5.1            C:\Program Files\Common Files... msi
```

It seems likely that some day we're going to have some helper methods for specifying the format of powershell commands and then parsing the output of the objects in that format, but that doesn't need to be today.